src/sentry/static/sentry/app/components/externalIssues/abstractExternalIssueForm.tsx

```diff
@@ -126,11 +131,17 @@ export default class AbstractExternalIssueForm<
     }
   };
 
+  /** For fields with dynamic fields, cache the fetched choices. */
   updateFetchedFieldOptionsCache = (
-    _field: IssueConfigField,
-    _result: {value: string; label: string}[]
+    field: IssueConfigField,
+    result: {value: string; label: string}[]
   ): void => {
-    // Do nothing.
+    const fetchedFieldOptionsCache = result.map(obj => [obj.value, obj.label]);
+    this.setState(prevState => {
```

Comment:
@mgaeta wondering why we need the callback version of `setState` instead of setting the state field directly.

Reply:
Should I set the state field directly like this? 
```
const {fetchedFieldOptionsCache} = this.state;
set(
  fetchedFieldOptionsCache,
  field.name,
  result.map(obj => [obj.value, obj.label])
);
this.setState({fetchedFieldOptionsCache});
```

Reply:
you should make a copy of `fetchedFieldOptionsCache` before calling `set` on it though. But at least we only have to clone a part of the state instead of the full state true.